src/Service.GraphQLBuilder/Sql/SchemaConverter.cs

```diff
@@ -130,7 +130,9 @@ public static class SchemaConverter
                         relationshipInfo.TargetEntityToFkDefinitionMap.TryGetValue(targetEntityName,
                             out List<ForeignKeyDefinition>? listOfForeignKeys))
                     {
-                        ForeignKeyDefinition? foreignKeyInfo = listOfForeignKeys.FirstOrDefault();
+                        ForeignKeyDefinition? foreignKeyInfo =
+                            listOfForeignKeys.Where(fk => fk.ReferencingColumns.Count > 0
```

Comment:
For 1:1 and n:1 relationships, the foreign key info is added in both the direction when iterating through the relationships defined in the config file - with the current entity as the referenced (and the related entity as referencing) as well as with current entity as the referencing entity (and the related entity as the referenced entity); as it is not possible to determine this without looking at the database metadata. Later when the database metadata is inferred, it is possible to know exactly the referenced/referencing entity, at which point, the referencing column and referenced column for exactly one pair (either current entity is referenced or current entity is referencing) would get updated. So, the entry with referencing/referenced column count = 0 is essentially invalid entry. 

Is this why, the condition (referencing & referenced columns count > 0) is introduced?

If so, adding the reasoning (in a better concise manner) would be helpful.

Reply:
yes, that's exactly the reason, will add it in comments